src/test/java/org/example/EchoServerTest.java

```diff
@@ -13,7 +13,7 @@
 class EchoServerTest {
 
     public EchoServer echoServerTest = new EchoServer();
-    public EchoSocket echoSocketTest = new EchoSocket();
+    public ClientSocket clientSocketTest = new ClientSocket();
 
     @Test
     public void testServerSocketGetsCreated() throws IOException {
```

Comment:
Good work in testing happy path. Need tests for when server socket and client socket connection fails. 

Hint: what error will it throw? check out [JUnit `assertThrows()`](https://www.baeldung.com/junit-assertions#junit5-assertThrows)

Reply:
Added tests for when server socket and client socket connection fails to latest commit. Thank you!